Fix build error with gcc14 due to missing strdup() declaration.
The definition of _POSIX_C_SOURCE hides the declaration in string.h,
so we need expliclty defining _XOPEN_SOURCE.

---
 liblash/socket.c |    1 +
 1 file changed, 1 insertion(+)

--- a/liblash/socket.c
+++ b/liblash/socket.c
@@ -18,6 +18,7 @@
  *   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#define _XOPEN_SOURCE 500 /* strdup() */
 #define _POSIX_C_SOURCE 200112L /* addrinfo */
 
 #include <stdint.h>
